docs/source/1.0/features/disagg-serving.md

```diff
@@ -1 +1,269 @@
+
+## Troubleshooting and FAQ
+
+### General FAQs
+
+*Q. What are the limitations of disaggregated serving in TRT-LLM?*
+
+A. Currently, only decoder-only models and beam width of 1  are supported. Also the KV cache at each layer of the model is required to be homogeneous, with the same data type and the same number of attention headers.
+
+*Q. When using the TRT backend, is the engine used for disaggregated serving different from other engines?*
+
+A. No. There are no special requirements for the arguments to build engine.
+
+*Q. When using the TRT backend, do the engines used by the context and generation instances need to be the same?*
+
+A. No. The engines used by context and generation instances can be different, and their parallelism can be heterogeneous, i.e., TP,PP can be different, and TRT-LLM will handle the heterogeneity of KV cache.
+
+*Q. Can a TRT-LLM server instance handle both context-only requests and generation-only requests?*
+
+A. Yes, but it's not recommended. TRT-LLM does not implement optimal scheduling for the case where the instance handles mixed context-only requests and generation-only requests. It's better to run context-only requests and generation-only requests on sets of servers.
+
+*Q. Does disaggregated serving in TRT-LLM support multi-gpu and multi-node?*
+
+A. Yes, it's recommended that different server instances use different GPUs. We support running context and generation servers on the same node or different nodes. The `CUDA_VISIBLE_DEVICES` env variable can be used to control which GPUs are used by each instance.
+
+*Q. What's the requirement for disaggregated serving in TRT-LLM?*
+
+A. TRT-LLM requires `UCX`-backend `CUDA-aware MPI` currently, TRT-LLM implements KV cache transfer with [`CUDA-aware MPI`](https://docs.open-mpi.org/en/v5.0.x/tuning-apps/networking/cuda.html#how-do-i-build-open-mpi-with-cuda-aware-support), and will support more communication components for KV cache transfer in future version.
+
+### Debugging FAQs
+
+*Q. How to handle error `Disaggregated serving is not enabled, please check the configuration?`*
+
+A. please set the environment variables
+```
+export TRTLLM_USE_MPI_KVCACHE=1
+```
+or
+```
+export TRTLLM_USE_UCX_KVCACHE=1
+```
+When the environment variable `TRTLLM_USE_MPI_KVCACHE=1` is set, TRT-LLM will transfer the KV cache using `CUDA-aware MPI`. All executor processes involved must share the same MPI world communicator. Consequently, with `TRTLLM_USE_MPI_KVCACHE=1`, TRT-LLM only supports launching multiple executors via `MPI`. Additionally, the `CommunicationMode` for the executors must be set to `kLEADER` or `kORCHESTRATOR` with `SpawnProcesses=false` for the `disaggregated-service`. These restrictions do not apply when `TRTLLM_USE_UCX_KVCACHE=1` is set.
```

Comment:
`SpawnProcesses=false`. I think we need more information on where this parameter should be set.